From 9cdaf6ea8d30b92017680e7a260fd127efdfb8c5 Mon Sep 17 00:00:00 2001
From: Matheos Worku <matheos.worku@sun.com>
Date: Wed, 5 Mar 2008 14:47:36 -0800
Subject: NIU: Fix BMAC alternate MAC address indexing.

From: Matheos Worku <matheos.worku@sun.com>

Upstream commit: 3b5bcedeeb755b6e813537fcf4c32f010b490aef

BMAC port alternate MAC address index needs to start at 1. Index 0 is
used for the main MAC address.

Signed-off-by: Matheos Worku <matheos.worku@sun.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>

---
 drivers/net/niu.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/net/niu.c
+++ b/drivers/net/niu.c
@@ -5148,7 +5148,12 @@ static void niu_set_rx_mode(struct net_d
 			index++;
 		}
 	} else {
-		for (i = 0; i < niu_num_alt_addr(np); i++) {
+		int alt_start;
+		if (np->flags & NIU_FLAGS_XMAC)
+			alt_start = 0;
+		else
+			alt_start = 1;
+		for (i = alt_start; i < niu_num_alt_addr(np); i++) {
 			err = niu_enable_alt_mac(np, i, 0);
 			if (err)
 				printk(KERN_WARNING PFX "%s: Error %d "
